src/devices/machine/i8291a.cpp

```diff
@@ -0,0 +1,1082 @@
+
+	switch (m_rl_state) {
+	case RemoteLocalState::LOCS:
+		break;
+	case RemoteLocalState::LWLS:
+		break;
+	case RemoteLocalState::REMS:
+		break;
+	case RemoteLocalState::RWLS:
+		break;
+	}
+}
+
+void i8291a_device::run_lp_fsm()
+{
+	switch(m_lp_state) {
+	case ListenerPrimaryState::LPIS:
+		m_address_status &= ~REG_ADDRESS_STATUS_LPAS;
+		break;
+	case ListenerPrimaryState::LPAS:
+		m_address_status |= REG_ADDRESS_STATUS_LPAS;
+		break;
+	}
+}
+
+void i8291a_device::run_fsm()
+{
+	if (m_ignore_ext_signals)
+		return;
+	m_ignore_ext_signals = true;
+	do {
+		m_state_changed = false;
+
+		auto m_l_state_old = m_l_state;
+		auto m_t_state_old = m_t_state;
+		auto m_rl_state_old = m_rl_state;
+		run_sh_fsm();
+		run_ah_fsm();
+		run_t_fsm();
+		run_tp_fsm();
+		run_tsp_fsm();
+		run_l_fsm();
+		run_lp_fsm();
+		run_dc_fsm();
+		run_dt_fsm();
+		run_sp_fsm();
+		run_pp_fsm();
+		run_rl_fsm();
+
+		if ((m_t_state_old == TalkerState::TIDS && m_t_state == TalkerState::TADS) ||
+				(m_t_state_old == TalkerState::TADS && m_t_state == TalkerState::TIDS) ||
+				(m_l_state_old == ListenerState::LIDS && m_l_state == ListenerState::LADS) ||
+				(m_l_state_old == ListenerState::LADS && m_l_state == ListenerState::LIDS))
+			m_ints2 |= REG_INTS2_ADSC;
+
+		if ((m_rl_state_old == RemoteLocalState::LOCS && m_rl_state == RemoteLocalState::REMS) ||
+				(m_rl_state_old == RemoteLocalState::REMS && m_rl_state == RemoteLocalState::LOCS) ||
+				(m_rl_state_old == RemoteLocalState::LWLS && m_rl_state == RemoteLocalState::RWLS) ||
+				(m_rl_state_old == RemoteLocalState::RWLS && m_rl_state == RemoteLocalState::LWLS))
+			m_ints2 |= REG_INTS2_REMC;
+
+		if ((m_rl_state_old == RemoteLocalState::LOCS && m_rl_state == RemoteLocalState::LWLS) ||
+				(m_rl_state_old == RemoteLocalState::REMS && m_rl_state == RemoteLocalState::RWLS) ||
+				(m_rl_state_old == RemoteLocalState::LWLS && m_rl_state == RemoteLocalState::LOCS) ||
+				(m_rl_state_old == RemoteLocalState::RWLS && m_rl_state == RemoteLocalState::REMS))
+			m_ints2 |= REG_INTS2_REMC;
+
+		m_rdy = !m_apt_flag && !m_cpt_flag && !m_din_flag;
+		update_int();
+	} while (m_state_changed);
+	m_ignore_ext_signals = false;
+}
+
+
+const char *i8291a_device::get_state_name(DeviceClearState state)
+{
+	switch(state) {
+		case DeviceClearState::DCIS:
+			return "DICS";
+		case DeviceClearState::DCAS:
+			return "DCAS";
+	}
+	return "Unknown";
+}
+
+const char *i8291a_device::get_state_name(DeviceTriggerState state)
+{
+	switch(state) {
+	case DeviceTriggerState::DTIS:
+		return "DTIS";
+	case DeviceTriggerState::DTAS:
+		return "DTAS";
+	}
+	return "Unknown";
+}
+
+
+const char *i8291a_device::get_state_name(ParallelPollState state)
+{
+	switch(state){
+	case ParallelPollState::PPIS:
+		return "PPIS";
+	case ParallelPollState::PPSS:
+		return "PPSS";
+	case ParallelPollState::PPAS:
+		return "PPAS";
+	}
+	return "Unknown";
+}
+
+
+const char *i8291a_device::get_state_name(SerialPollState state)
+{
+	switch(state) {
```

Comment:
Please be consistent with spacing - you're using `if (...` but `switch(...` - they're both flow control constructs, so there's no reason to use different spacing for them.